### PR TITLE
Add 'redir' href with parameters to search and list results

### PR DESCRIPTION
**Pull request by Crispioso**
### What

Add a redirected URL into the href for search and list page results, so that we can get some stats on what results are being selected by people for what terms (links in with https://github.com/ONSdigital/dp-frontend-router/tree/feature/search-stats).

### How to review

Search for something (with router switched to branch `feature/search-states`) and check that it's first going to a path beginning with 'redir' and all the parameters have a value that looks correct.

### Who can review

Anyone but me.

### CODE REVIEW

File: src/main/web/templates/handlebars/list/partials/results-title.handlebars [hunk 1]
```diff
@@ -1,3 +1,3 @@
-{{!-- Removed indentation to remove space between title and edition --}}
 <h3 class="search-results__title">
-    <a href="{{uri}}">{{{description.title}}}{{#if_all description.edition showEdition}}{{#if_all (ne description.edition 'yes') (ne description.edition '<strong>yes</strong>')}}: {{{description.edition}}}{{/if_all}}{{/if_all}} {{#if description.latestRelease}}({{labels.latest-release}}){{/if}}</a></h3>
\ No newline at end of file
+        <a href="/redir?url={{uri}}&term={{parameters.query.0}}{{parameters.q.0}}&pageIndex={{result.paginator.currentPage}}&linkIndex={{@index_1}}&type={{listType}}&pageSize={{#if parameters.size.0}}{{parameters.size.0}}{{else}}10{{/if}}">{{{description.title}}}{{#if_all description.edition showEdition}}{{#if_all (ne description.edition 'yes') (ne description.edition '<strong>yes</strong>')}}: {{{description.edition}}}{{/if_all}}{{/if_all}} {{#if description.latestRelease}}({{labels.latest-release}}){{/if}}</a>
```
Comment (CarlHembrough) [inline src/main/web/templates/handlebars/list/partials/results-title.handlebars]: Not sure that '{{parameters.query.0}}' is required for the term field? when debugging I had only a value for '{{parameters.q.0}}'.

Comment (ian-kent) [inline src/main/web/templates/handlebars/list/partials/results-title.handlebars]: apparently one is for search and the other is for list pages 😸

File: src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * Created by iankent on 03/02/17.
+ */
+public enum SignedURLHelper implements BabbageHandlebarsHelper<String> {
+
+    //Sign a URL
+    sign_url {
+        @Override
+        public CharSequence apply(String context, Options options) throws IOException {
+            // (sign_url uri parameters.query.0 parameters.q.0 result.paginator.currentPage @index_1 listType parameters.size.0)
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java]: delete commented out code.

File: src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Created by iankent on 03/02/17.
+ */
+public enum SignedURLHelper implements BabbageHandlebarsHelper<String> {
+
+    //Sign a URL
+    sign_url {
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java]: Enum should be all upper case.

File: src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            if (options.param(5) == null) {
+                pageSize = 10;
+            } else {
+                pageSize = Integer.valueOf(options.param(5));
+            }
+
+            String token = "";
+            try {
+                token = JWT.create()
+                        //.withIssuer("auth0")
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java]: delete commented out code.

File: src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java (hunk 3 above)
Comment (ian-kent) [reply src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java]: don't think we can for these, they're actually template functions and it'd be inconsistent with all the existing helpers

Comment (daiLlew) [reply src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/SignedURLHelper.java]: fine.

**Pull request merged**